Data/DataModel/HelpMessage.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace ServerCore.DataModel
+{
+    /// <summary>
+    /// Message in help thread when a player asks for help.
+    /// </summary>
+    public class HelpMessage
+    {
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int ID { get; set; }
+
+        /// <summary>
+        /// The date time (in UTC) of when the message was sent.
+        /// </summary>
+        public DateTime DateTimeInUtc { get; set; }
+
+        /// <summary>
+        /// The puzzle id that this message is for.
+        /// </summary>
+        [Required]
+        public int PuzzleID { get; set; }
+
+        /// <summary>
+        /// The puzzle this is a help message for.
+        /// </summary>
+        [Required]
+        public virtual Puzzle Puzzle { get; set; }
+
+        /// <summary>
+        /// The actual message text.
+        /// </summary>
+        public string Text { get; set; }
+
+        /// <summary>
+        /// True if message was from the player (as opposed to a response from puzzle control).
+        /// </summary>
+        public bool IsPlayerMessage { get; set; }
+
+        /// <summary>
+        /// The team ID of the sender (will be empty if not a player of if it is for a SinglePlayerPuzzle).
+        /// </summary>
+        public int TeamID { get; set; }
+
+        /// <summary>
+        /// The team of the sender (will be empty if not a player or if it is for a SinglePlayerPuzzle).
+        /// </summary>
+        public virtual Team Team { get; set; }
+
+        /// <summary>
+        /// The id of the sender.
+        /// </summary>
+        public int SenderID { get; set; }
```

Comment:
Ditto my comment above, and in addition I think that when authors reply, we should not tell the player who has replied. No problem reporting which player has posted a team message though.

Reply:
We will likely not tell the player who replied in the UI (will show up just as puzzle control) but it might be nice information to have on our side. It will also make it so that the model is generally the same

Reply:
EventId, PuzzleId, and TeamID is there.
Do we really need staffId or playerId? Wouldn't the PuzzleUserId be enough? SinglePlayerPuzzles can be solved even if you aren't a registered player